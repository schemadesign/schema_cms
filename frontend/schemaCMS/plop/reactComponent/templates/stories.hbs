import React from 'react';
import { storiesOf } from '@storybook/react';

import { {{ pascalCase name }} } from './{{ camelCase name }}.component';

export const defaultProps = {};

storiesOf('{{ pascalCase name }}', module).add('Default', () => <{{ pascalCase name }} {...defaultProps} />);
