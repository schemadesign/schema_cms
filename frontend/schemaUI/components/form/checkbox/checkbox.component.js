import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { getStyles } from './checkbox.styles';
import CheckboxGroupContext from '../checkboxGroup/checkboxGroup.context';
import { EditIcon } from '../../icons/editIcon';
import { withStyles } from '../../styles/withStyles';

export class CheckboxComponent extends PureComponent {
  static propTypes = {
    id: PropTypes.string.isRequired,
    label: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
    value: PropTypes.any.isRequired,
    isEdit: PropTypes.bool,
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  };

  renderCheckboxIcon = ({ checked, checkedIcon, uncCheckedIcon }) => (checked ? checkedIcon : uncCheckedIcon);

  renderEditIcon = isEdit => (isEdit ? <EditIcon /> : null);

  render() {
    const { id, children, isEdit, theme, ...restProps } = this.props;

    return (
      <CheckboxGroupContext.Consumer>
        {({ onChange, value = [], name, checkedIcon, uncCheckedIcon, customCheckboxStyles }) => {
          const checked = value.includes(restProps.value);
          const { containerStyles, inputStyles, iconContainerStyles, labelStyles, elementStyles } = getStyles(theme);
          const styles = { ...containerStyles, ...customCheckboxStyles };

          return (
            <div style={styles}>
              <input
                {...restProps}
                onChange={onChange}
                value={restProps.value}
                name={name}
                checked={checked}
                aria-hidden
                type="checkbox"
                id={id}
                style={inputStyles}
              />
              <span style={elementStyles}>{children}</span>
              <div style={iconContainerStyles}>
                {this.renderEditIcon(isEdit)}
                <label style={labelStyles} htmlFor={id}>
                  {this.renderCheckboxIcon({ checked, checkedIcon, uncCheckedIcon })}
                </label>
              </div>
            </div>
          );
        }}
      </CheckboxGroupContext.Consumer>
    );
  }
}

export const Checkbox = withStyles(CheckboxComponent);
