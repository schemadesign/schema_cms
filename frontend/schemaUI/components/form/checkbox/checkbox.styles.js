import { defaultTheme } from '../../../utils/theme';

export const getStyles = (theme = defaultTheme) => ({
  containerStyles: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: 48,
    color: theme.checkbox.text,
    borderBottom: `1px solid ${theme.checkbox.border}`,
  },
  iconContainerStyles: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
  inputStyles: {
    visibility: 'hidden',
    position: 'absolute',
    overflow: 'hidden',
    height: 0,
    top: 0,
    left: 0,
  },
  labelStyles: {
    cursor: 'pointer',
  },
});
