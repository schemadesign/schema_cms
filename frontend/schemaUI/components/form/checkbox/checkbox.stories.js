import React from 'react';
import { storiesOf } from '@storybook/react';

import { Checkbox } from './checkbox.component';

const defaultProps = {};

storiesOf('Checkbox', module).add('Default', () => <Checkbox {...defaultProps} />);
