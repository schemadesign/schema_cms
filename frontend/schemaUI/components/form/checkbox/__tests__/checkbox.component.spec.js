import React from 'react';
import { shallow } from 'enzyme';

import { Checkbox } from '../checkbox.component';
import { defaultProps, context } from '../checkbox.stories';
import CheckboxGroupContext from '../../checkboxGroup/checkboxGroup.context';

describe('RadioButton: Component', () => {
  const component = props => (
    <CheckboxGroupContext.Provider value={context}>
      <Checkbox {...defaultProps} {...props}>
        radio
      </Checkbox>
    </CheckboxGroupContext.Provider>
  );

  const render = (props = {}) =>
    shallow(component(props))
      .dive()
      .dive();

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
