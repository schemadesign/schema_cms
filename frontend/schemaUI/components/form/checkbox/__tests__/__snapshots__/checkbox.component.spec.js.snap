// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton: Component should render checked 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "borderBottom": "1px solid #F0EFEF",
      "display": "flex",
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "space-between",
    }
  }
>
  <input
    aria-hidden={true}
    checked={true}
    id="id"
    name="name"
    onChange={[Function]}
    style={
      Object {
        "height": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "visibility": "hidden",
      }
    }
    type="checkbox"
    value="value"
  />
  <span>
    radio
  </span>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <EditIcon />
    <label
      htmlFor="id"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <CheckboxOnIcon />
    </label>
  </div>
</div>
`;

exports[`RadioButton: Component should render correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "borderBottom": "1px solid #F0EFEF",
      "display": "flex",
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "space-between",
    }
  }
>
  <input
    aria-hidden={true}
    checked={false}
    id="id"
    name="name"
    onChange={[Function]}
    style={
      Object {
        "height": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "visibility": "hidden",
      }
    }
    type="checkbox"
    value="value 2"
  />
  <span>
    radio
  </span>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <EditIcon />
    <label
      htmlFor="id"
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <CheckboxOffIcon />
    </label>
  </div>
</div>
`;
